nupic/swarming/hypersearch/utils.py

```diff
@@ -753,3 +754,25 @@ def sortedJSONDumpS(obj):
 
   else:
     return json.dumps(obj)
+
+
+
+def matchPatterns(patterns, keys):
```

Comment:
Is this duplicated from somewhere else?  Please include a TODO or something in the docstring that indicates that it was copied from nupic so that it may be refactored.

Otherwise, I have some feedback for how this ought to change.  For example, I don't believe the docstring doesn't matches the doxygen format, or sphinx (commonly used for documenting python code).  The fact that it returns `None` or a `list()` (which may be empty) doesn't seem right.